src/platform/cannonlake/include/platform/memory.h

```diff
@@ -245,10 +245,29 @@
 #define HEAP_HP_BUFFER_COUNT \
 			(HEAP_HP_BUFFER_SIZE / HEAP_HP_BUFFER_BLOCK_SIZE)
 
+#if defined(CONFIG_DMIC)
+#define SOF_TEXT_DMIC_SIZE 0x2000
+#else
+#define SOF_TEXT_DMIC_SIZE 0
+#endif
+
+#if defined(CONFIG_VOLUME)
+#define SOF_TEXT_VOLUME_SIZE 0x1000
+#else
+#define SOF_TEXT_VOLUME_SIZE 0
+#endif
+
+#if defined(CONFIG_SRC)
+#define SOF_TEXT_SRC_SIZE 0x1000
+#else
+#define SOF_TEXT_SRC_SIZE 0
+#endif
+
 /* text and data share the same HP L2 SRAM on Cannonlake */
 #define SOF_TEXT_START		0xBE040400
 #define SOF_TEXT_BASE		(SOF_TEXT_START)
-#define SOF_TEXT_SIZE		(0x1b000 - 0x400 + 0x2000)
+#define SOF_TEXT_SIZE		(0x17000 - 0x400 + 0x2000 + SOF_TEXT_DMIC_SIZE \
```

Comment:
Too many magic numbers,  would be better to replace `0x17000 - 0x400 + 0x2000` with something like `SOF_TEXT_BASE_SIZE`

Reply:
@lgirdwood 
TBH we can make our lives easier and just remove SOF_TEXT_SIZE. We can make it build minimal size always automatically. Is there any reason why we have to play with these defines? Also only SOF_TEXT_SIZE seems to work with rimage, defines for .data and .bss have no effect on final binary size (id if it's bug) and it's always minimal size (for example when you just remove src coefs tables from the build, it has smaller .data section with no changes to SOF_DATA defines).

Reply:
Removed magic numbers (added new SOF_TEXT_MIN_SIZE).
Also @lgirdwood  I can confirm that SOF_DATA_SIZE have no effect, we can set it to 10 and it will be okay. On the other hand SOF_TEXT_SIZE is working as expected so in the last commit I have added calculations for the minimum size selected. But the best it would be if it can build with minimum size automatically